internal/handler/wallet/backup.go

```diff
@@ -25,23 +25,27 @@ func (w *walletBackupAccount) Handle(params operations.BackupAccountParams) midd
 		return resp
 	}
 
-	go handleBackup(wlt, w.prompterApp)
-
-	return operations.NewDeleteAccountNoContent()
-}
-
-func handleBackup(wlt *wallet.Wallet, prompterApp prompt.WalletPrompterInterface) {
 	promptRequest := prompt.PromptRequest{
 		Action: walletapp.Backup,
 		Msg:    fmt.Sprintf("Backup wallet %s:", wlt.Nickname),
 		Data:   nil,
 	}
-
-	_, err := prompt.WakeUpPrompt(prompterApp, promptRequest, wlt)
+	promptOutput, err := prompt.WakeUpPrompt(w.prompterApp, promptRequest, wlt)
 	if err != nil {
-		return
+		return operations.NewBackupAccountUnauthorized().WithPayload(
+			&models.Error{
+				Code:    errorCanceledAction,
+				Message: "Unable to backup wallet",
+			})
+	}
+
+	if promptOutput != nil {
```

Comment:
it is not really readable, it checks if the output is not nil and it means that the user requested private key.